### DISCUSSION TITLE
How do you guys handle version control and CI/CD?

### DISCUSSION THREAD

**Discussion by falko100** [directus/directus#17269] [Archive]
Hey all!

We recently obtained to clients who use self-hosted Directus as a backend.
So far, everything looks amazing and the small changes we've done were easy to implement.

I ran into a couple of things and thought: let's ask the community.

1. Who do you guys handle version control for the following things;
    1. Schema / Collection information. I managed to get everything into one .yml file, but I would love to be able to alter the structure using multiple yml files and perhaps some migrations for moving data between columns when changes happen.
    2. Flows / Automations. I can't find and way of doing this so far, apart from database exports. Is there any way in Directus core for this?
2. Documentation references a custom docs page for our specific schema, yet I cannot find this. Is it deprecated or hidden behind some config setting?
3. I'd like to keep my extensions in the same GitHub repo as my schema. Are there any resources available about deploying this to Digital Ocean from one repo (with a docker-compose.yml)? 

I'd love to hear your thoughts and ideas on this!

**Comment by EdouardDem** [comment]
Hi, I've developped a CLI for this purpose: https://github.com/tractr/directus-sync
It allows to **pull** all schema, dashboards, flows, folders, operations, panels, permissions, presets, roles, settings, translations, and webhooks into JSON files (in your repository), and then **push** them back to another Directus instance.
This is a non-destructive operation and only apply changes.
I have also written a tutorial : https://medium.com/tractr/streamlining-directus-workflows-with-directus-sync-a-comprehensive-guide-baf33b9cc399
Hope this helps
